dot-net-sdk/store/ConfigurationStore.cs

```diff
@@ -94,19 +95,42 @@ private void ClearCaches()
     {
         _flagConfigurationCache.Clear();
         _banditModelCache.Clear();
-        _banditFlagCache.Clear();
+        _metadataCache.Clear();
     }
     public void LoadConfiguration()
     {
-        FlagConfigurationResponse flagConfigurationResponse = FetchFlags();
-        BanditModelResponse banditModels = FetchBandits();
-        SetConfiguration(
-            flagConfigurationResponse.Flags.ToList().Select(kvp => kvp.Value),
-            flagConfigurationResponse.Bandits,
-            banditModels.Bandits?.ToList().Select(kvp => kvp.Value));
+        // Get the last tag for flags.
+        string? etag = GetLastFlagVersion();
 
+        var flagConfigurationResponse = FetchFlags(etag);
+        if (flagConfigurationResponse.IsModified)
```

Comment:
so we're letting the eTag work flow upstream to the HTTP request layer, and just keying of `IsModified`? 

It would be useful to include more code comments detailing what is happening and what would cause IsModified to be true or false

Reply:
Agreed. Added a number of comments at key points.